src/App.tsx

```diff
@@ -1,24 +1,106 @@
-import React, { FC } from 'react';
-import logo from './logo.svg';
-import './App.css';
+import React, { FC, useState } from 'react';
+import {
+  Box,
+  Button,
+  Container,
+  CssBaseline,
+  TextField,
+} from '@material-ui/core';
+import Todo from './components/Todo/Todo';
+import { Todo as ITodo } from './components/Todo/typings';
 
-const App: FC = () => (
-  <div className="App">
-    <header className="App-header">
-      <img src={logo} className="App-logo" alt="logo" />
-      <p>
-        Edit <code>src/App.tsx</code> and save to reload.
-      </p>
-      <a
-        className="App-link"
-        href="https://reactjs.org"
-        target="_blank"
-        rel="noopener noreferrer"
+const App: FC = () => {
+  const [todoList, setTodoList] = useState<ITodo[]>([]);
+  const [newTodoName, setNewTodoName] = useState<string>('');
+  const [inputTodoError, setInputTodoError] = useState<string>('');
+
+  const changeTodoNameHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTodoName(event.target.value);
+  }
+
+  const addTodoHandler = () => {
+    const hasNewTodoAnInvalidName = !newTodoName || newTodoName.length < 6 || newTodoName.length > 20;
+    if (hasNewTodoAnInvalidName) {
+      setInputTodoError('Entrada inválida (digite de 6 a 20 caracteres)');
+      return;
+    } 
+    const newTodo: ITodo = {
+      title: newTodoName,
```

Comment:
I suggest making this code consistent, ex:
```suggestion
      name: newTodoName,
```

Reply:
Resolved on a0289c16fcae438d2f942c37bdf5ac34978d3349 and on ac40b7bd5b1418375c16876812c614ac7cb7f273